pkg/controller/migration/convert/core.go

```diff
@@ -211,210 +467,372 @@ func checkNodeHostPathVolume(spec corev1.PodSpec, name, path string) error {
+					Name:      render.CalicoNodeObjectName,
+					Resources: compRes.ResourceRequirements,
+				}
+				install.Spec.CalicoNodeDaemonSet.Spec.Template.Spec.Containers = append(install.Spec.CalicoNodeDaemonSet.Spec.Template.Spec.Containers, container)
+			}
+		case operatorv1.ComponentNameTypha:
+			// Ensure the override field is non nil
+			helpers.EnsureTyphaContainersNotNil(install)
+
+			// If the container already exists, do nothing since this container override takes precedence.
+			var found bool
+			for _, c := range install.Spec.TyphaDeployment.Spec.Template.Spec.Containers {
+				if c.Name == render.TyphaContainerName {
+					found = true
+				}
+			}
+
+			// If the container is not already specified, create the override container entry for it.
+			if !found {
+				container := operatorv1.TyphaDeploymentContainer{
+					Name:      render.TyphaContainerName,
+					Resources: compRes.ResourceRequirements,
+				}
+				install.Spec.TyphaDeployment.Spec.Template.Spec.Containers = append(install.Spec.TyphaDeployment.Spec.Template.Spec.Containers, container)
+			}
+		case operatorv1.ComponentNameKubeControllers:
+			// Ensure the override field is non nil
+			helpers.EnsureKubeControllersContainersNotNil(install)
+
+			// If the container already exists, do nothing since this container override takes precedence.
+			var found bool
+			for _, c := range install.Spec.CalicoKubeControllersDeployment.Spec.Template.Spec.Containers {
+				if c.Name == kubecontrollers.KubeController {
+					found = true
+				}
+			}
+
+			// If the container is not already specified, create the override container entry for it.
+			if !found {
+				container := operatorv1.CalicoKubeControllersDeploymentContainer{
+					Name:      kubecontrollers.KubeController,
+					Resources: compRes.ResourceRequirements,
+				}
+				install.Spec.CalicoKubeControllersDeployment.Spec.Template.Spec.Containers = append(install.Spec.CalicoKubeControllersDeployment.Spec.Template.Spec.Containers, container)
+			}
+		}
 	}
-	if reflect.DeepEqual(existingRR, rescReq) {
+
+	// Lastly, clear out the ComponentResources slice.
+	install.Spec.ComponentResources = nil
+}
+
+// migrateContainerResources adds the resources for the specified component container if none was previously set. If the Installation
+// already had a resource for the component container then they are compared and if they are different then an error is returned.
+// If the component container resource is added to the Installation or the existing one matches then nil is returned.
+func migrateContainerResources(install *operatorv1.Installation, componentName, containerName string, resources *corev1.ResourceRequirements) error {
+	// If resources already exist for this component container, verify that they equal the container's existing resources.
+	if existingResources, found := getContainerResourceOverride(install, componentName, containerName); found {
+		if !reflect.DeepEqual(existingResources, resources) {
```

Comment:
I need to address this. This is wrong since slice element order is taken into account in reflect.DeepEqual.